src/main/java/ee/buerokratt/ruuter/domain/steps/ReturnStep.java

```diff
@@ -18,6 +19,10 @@ public class ReturnStep extends ConfigurationStep {
     @Override
     protected void executeStepAction(ConfigurationInstance ci) {
         ci.setReturnValue(ci.getScriptingHelper().evaluateScripts(returnValue, ci.getContext(), ci.getRequestBody(), ci.getRequestParams()));
+        Integer finalResponseStatusCode = ci.getProperties().getFinalResponse().getHttpStatusCode();
+        if (finalResponseStatusCode != null && ci.getReturnValue() instanceof HttpStepResult ciReturnValue) {
```

Comment:
I think it would be better to move this logic into the controller (since the controller decides the end status). You could easily read out the application property and then, if it exists, assign a custom response status.